WorldBrain/Memex
# [WIP] Sync functionality needed for MVP

This includes first steps in refactoring for Sync.

To test before merging:
- [ ] Import
- [ ] Backup and restore

Although it looks big, this is actually a quite boring refactor. Should break things in obvious ways if things go wrong.

One important thing to note though are the changes to `tsconfig.json` to allow for synthetic default imports. This means that instead of doing `import * as expect from 'expect'` you do `import expect from 'expect'`. This should iron out some inconsistencies with importing, which is quite a pain.
